hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/XMLUtils.java

```diff
@@ -156,8 +156,18 @@ public static SAXTransformerFactory newSecureSAXTransformerFactory()
           throws TransformerConfigurationException {
     SAXTransformerFactory trfactory = (SAXTransformerFactory) SAXTransformerFactory.newInstance();
     trfactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
-    trfactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
-    trfactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "");
+    quietSet(trfactory, XMLConstants.ACCESS_EXTERNAL_DTD, "");
+    quietSet(trfactory, XMLConstants.ACCESS_EXTERNAL_STYLESHEET, "");
     return trfactory;
   }
+
+  private static boolean quietSet(TransformerFactory transformerFactory, String name, Object value) {
+    try {
+      transformerFactory.setAttribute(name, value);
+      return true;
+    } catch (Exception|Error e) {
+      // ok to ignore
```

Comment:
Can we record some logs?

Reply:
these logs can be noisy - when I've added similar code in Apache POI and elsewhere, users gripe about the same lines appearing in the log even if it isn't exactly a storm - could they be debug level logs?